### PR TITLE
Moved 5 green micro-optimization into solver logic

### PR DESCRIPTION
**Pull request by ftommasi**
title

### CODE REVIEW

File: play.py [hunk 1]
```diff
@@ -57,15 +57,19 @@ def play(mode, answer, valid_words, hard, rounds, debug=False):
...
         if round_number == DEFAULT_ROUNDS + 1 and mode in [Mode.PLAY, Mode.SOLVE]:
-            print('---------------')
+            print('---------------')    
             
         if state and debug:
             print(f'total green: "{"".join(state.green)}"')
             print(f'total yellow: {sorted(state.yellow)}')
             print(f'total gray: {sorted(list(state.grey))}')
 
+       #if we know all green spots, then there is no need to keep guessing
```
Comment (cwlucas41) [inline play.py]: nit: remove commented code

File: play.py [hunk 2]
```diff
@@ -57,15 +57,19 @@ def play(mode, answer, valid_words, hard, rounds, debug=False):
     state = None
     for round_number in range(1, rounds + 1):
         if round_number == DEFAULT_ROUNDS + 1 and mode in [Mode.PLAY, Mode.SOLVE]:
-            print('---------------')
+            print('---------------')
```
Comment (cwlucas41) [inline play.py]: nit: remove whitespace change

**Review by cwlucas41 — APPROVED** [review]
please fix nits

**Comment by ftommasi** [comment]
nits fixed

**Pull request merged**